src/processFile.ts

```diff
@@ -8,11 +8,11 @@ import { readSourceFile, SOURCE_MAP_URL_MARKER } from './readSourceFile.js';
 import type { SourceFile, SourceMap } from './SourceFile.js';
 import { UsageError } from './errors.js';
 
-const isSupportedFileMJS = /\.(m?js|d\.m?ts)$/;
-const isSupportedFileCJS = /\.(c?js|d\.c?ts)$/;
+const isSupportedFileMJS = /\.(m?js|m?ts|d\.m?ts)$/;
+const isSupportedFileCJS = /\.(c?js|c?ts|d\.c?ts)$/;
 
 const regExpImportExport = /(import|export).*? from ('|")(?<file>\..*?)\2;/g;
```

Comment:
```suggestion
const regExpImportExport = /(import|export).*? from ('|")(?<file>\..*?)\2;/dg;
```
I wasn't aware of the `d` flag when this code was written. Using it will make things easier.